### DISCUSSION TITLE
Can Faiking set the demand control of the outdoor unit?

### DISCUSSION THREAD

**Discussion by MUN0X** [revk/xESP32-Faikout#179] [Q&A]
Hi!
I currently run a daikin multi-split with 4 units connected to home assistant
3 of them are running on the BRP069B4x module, the 4th one has a BRP069C4x module.
I ordered a faikin module that will replace the C-module to get rid of the cloud dependency.

However the integrated controller with an external BLE sensor makes me wonder whether I should replace my B-modules as well.
I currently have several automations running in Home assistant to change the setpoints of the daikin units based on external temperature sensors. However, it would be nice to have the standalone system of faikin in case Home assistant goes down (or my automation f*cks-up 😊). One of my automations limits the power draw of the outdoor unit depending on the outside temperature to get a smoother temperature control (essentially forcing the compressor frequency down, mostly to 15-20hz). I do this via the “set_demand_control” command in the daikin api.

Is it possible to do this via the Faikin controller as well?
I looked through the manual and the discussions but couldn't find a clear answer

**Comment by revk** [comment]
I have not made any system to link the Faikin together to do something like that - obviously it may be possible, but generally I have assumed that such integration would be done by HA or similar.

But if the Daikin API has it, it may be worth considering - do you know what that actually does in practice?

**Comment by MUN0X** [comment]
I'm sorry if I didn't make myself clear. 
My goal is not to link the faiking modules together and agree that you should use home assistant for that.
Having home assistant set a desired setpoint for the automated controls of each faikin module and then letting the automated controls of the faikin modules do their (independent) thing is exactly what I'm after.

The thing is that the genuine daiking modules (BRP069B4x) allow me to send a REST command to the outdoor unit to limit its power draw.
By doing so i make the outdoor unit less "aggressive" and make the temperature control of the indoor unit much more gradual. 
This is especially useful when is it for heating with only 1 or 2 out of 4 indoor units switched on. Without this restriction one active indoor unit would turn the outdoor unit off/on with 5-10min intervals where the outdoor unit would draw ~2000W intermittently.  By limiting the power draw of the outdoor unit down to approx ~500W this behavior stops and the outdoor unit will stay on for 30+min at a time. This is much more comfortable and improve the efficiency of the system quite a bit.

To do this i pass the following command to my B-modules. They seem to pass it on to the outdoor unit because the C-unit, which i can’t control via the local API, respects the imposed restrictions.
`http://IP/aircon/set_demand_control?lpw=&mode=0&suc=0&moc=0&tuc=0&wec=0&sac=0&en_demand=1&frc=0&max_pow=40&type=1&thc=0`

Based on my own testing the parameters are the following:
-En_demand:
	0 = Restrictions off
	1 = Restriction on	
-Mode:
	0 = manual restriction
	1 = unknown, but I suspect this enable the schedule set by the days
	2 = automatic, not sure how it works exactly. it respects a limit at some times, does it’s own thing on other occasions (doesn’t work well)
-Max_power:  max power (%) of the outdoor unit.
 Any integer between 40 en 100 seems to be accepted.
-Moc, tuc,…, suc = weekdays, I guess to set a weekly schedule but I’ve never used it.
-Type = unknown, default is 1

To be clear you can also set these restriction via the app but using the REST commands via home assistant makes it much more useful.
The default demand control presets in the (old) app translate to:
OFF = `http://IP/aircon/set_demand_control?lpw=&mode=0&en_demand=0&max_pow=70&type=1`
LOW = `http://IP/aircon/set_demand_control?lpw=&mode=0&en_demand=1&max_pow=40&type=1`
MED = `http://IP/aircon/set_demand_control?lpw=&mode=0&en_demand=1&max_pow=50&type=1`
HIGH = `http://IP/aircon/set_demand_control?lpw=&mode=0&en_demand=1&max_pow=70&type=1`
AUTO = `http://IP/aircon/set_demand_control?lpw=&mode=2&en_demand=1&max_pow=85&type=1`

**Reply by revk** [reply]
OK, basically we need to find the S21 messages that result from this setting. If we can find that, then easy to add.

This means dumping the serial messages from that controller.

**Comment by MUN0X** [comment]
Just to illustrate what I mean: 
![image](https://github.com/revk/ESP32-Faikin/assets/78272516/c16016dc-47fb-443f-8a1d-4c403e13386c)

This is with 2 - 4 units active throughout the last 24hrs. I switched the max_power value manually at certain times to illustrate the behavior. 
In normal operation it would have stayed between 50% and 70%.
As you can see my heating was initially running fine at 50%. Running constantly at a power draw of ~700w with a comp freq of 25hz.
By lowering the max_pow to 40 these numbers dropped to ~500w and 20 hz. This was too low with these outside temperatures and the temperature in my largest room started to drop.
I then increased the power back to 50%, then 70% and finally to 100% by disabling the en_demand function.
As I increase the power limit you can see the power draw and frequency becoming erratic as the indoor temperatures approach and surpass their setpoint (20°C). Finally the outdoor unit switches off until the room temperatures drop below 19.5°C again. 
These temperature swings and on/off behavior gets much worse with only 1 room active and mild temperatures outside (it was -8°C outside) 

hope this helps to explain what I'm trying to achieve :)

**Comment by JeanOlivier** [comment]
Just to add here that this feature would be very convenient. My units tend to run the compressor slow and they can't keep up. Being able to ask the compressor to be more aggressive would be very helpful.

@MUN0X if you can debug this that'd be great.

**Reply by revk** [reply]
We just need to know what the lower level S21 messages are for this.

**Reply by MUN0X** [reply]
How can i dump these? Is there a guide or procedure somewhere?

**Reply by revk** [reply]
You need something to connect to the serial. It may be possible to wire up a Faikin with serial rx tapped on to the serial line and fix the protocol and set debug packet dumping. That may be complicated.

I am wondering if I aloud make a monitoring device with plug and socket to go in line for such cases...

**Reply by MUN0X** [reply]
So connect gnd, pwr and rx of the faikin tapped into the lines between the existing module an the ac? The rx connected with the tx of the module presume? 

I have a ch340 programmer lying around somewhere. Could that be of help?

**Reply by revk** [reply]
Basically. The S21 protocol can be fixed by a setting and dump enabled.

**Reply by MUN0X** [reply]
Allright, i recieved the faikin and will give it a shot

**Reply by MUN0X** [reply]
I'm afraid that i will need some more help with this.
So I used the following settings from one of your earlier posts: "set protofix to 1, protocol to 8, dump to 1" 
I guess the protocol depends on the daiking unit. 2,3 and 4 appear to be related to S21. How do I know which one to pick?

just to be sure, the Rx pin of the faikin module is the one labelled "34", right?

**Reply by revk** [reply]
The first 4 (0,1,2,3) should be S21 (off the top of my head). Two are interred tx and two are inverted rx. I would expect 0 to be what you want.

If you see rx of anything with dump enabled you have the right pin.

**Reply by MUN0X** [reply]
Allright, I finally managed to find some time to try again.
With protocol set to 0, I seem to get swamped by rx messages.
However the info/faikin/s21 is an empty JSON 99% of the time. 

How do I make sense of this?

**Reply by revk** [reply]
It will say loopback if what it receives is exactly what it sent, it does not really know it is just monitoring. I could make a setting to simplify this.

If you just monitor the rx messages you should see they repeat a sequence of commands.

Then you can change a setting and find which of these changes for the setting you changed.

It is a pain, I know.

**Reply by revk** [reply]
I'll issue code with a `snoop` setting, this will not `tx` but will dump what it sees. Should be issued in a few minutes.

**Reply by MUN0X** [reply]
Alright, I seem to be getting somewhere!
The snoop function makes it a whole lot easier. Much less chatter on the rx line :)

The Demand_control appears to be set by D7. 
The original daikin modules takes the folowing command: `http://IP/aircon/set_demand_control?lpw=&mode=0&suc=0&moc=0&tuc=0&wec=0&sac=0&en_demand=1&frc=0&max_pow=40&type=1&thc=0` with only `mode` `en_demand` `max_pow` being mandatory for the command to be accepted.

Out of all those variables only changes to `en_demand` `max_pow` appear to trigger a D7 message via s21.
I guess this makes sense since all other parameter appear to be related to some sort of scheduling feature (telling the original daikin module when to send a D7 message to the AC?).

As for the messages being sent to the AC:
En_demand | Max_pow | S21
0 | 40 | {"protocol":"S21","dump":"024437303030303B03","D7":"0000"}
0 | 80 | {"protocol":"S21","dump":"024437303030303B03","D7":"0000"}
0 | 100 | {"protocol":"S21","dump":"024437303030303B03","D7":"0000"}
1 | 100 | {"protocol":"S21","dump":"024437303030303B03","D7":"0000"}
1 | 99 | {"protocol":"S21","dump":"024437313030303C03","D7":"1000"}
1 | 95 | {"protocol":"S21","dump":"024437353030304003","D7":"5000"}
1 | 90 | {"protocol":"S21","dump":"0244373A3030304503","D7":":000"}
1 | 85 | {"protocol":"S21","dump":"0244373F3030304A03","D7":"?000"}
1 | 80 | {"protocol":"S21","dump":"024437443030304F03","D7":"D000"}
1 | 75 | {"protocol":"S21","dump":"024437493030305403","D7":"I000"}
1 | 70 | {"protocol":"S21","dump":"0244374E3030305903","D7":"N000"}
1 | 65 | {"protocol":"S21","dump":"024437533030305E03","D7":"S000"}
1 | 60 | {"protocol":"S21","dump":"024437583030306303","D7":"X000"}
1 | 55 | {"protocol":"S21","dump":"0244375D3030306803","D7":"]000"}
1 | 50 | {"protocol":"S21","dump":"024437623030306D03","D7":"b000"}
1 | 45 | {"protocol":"S21","dump":"024437673030307203","D7":"g000"}
1 | 40 | {"protocol":"S21","dump":"0244376C3030307703","D7":"l000"}

Does that help you?

**Reply by MUN0X** [reply]
I just tried to send `D7l000` (=40%) via the command/Faikin/send functionality and the unit reduces its cmpfreq/power draw as expected.
Sending D70000 makes the compressor ramp back up to max.

**Reply by revk** [reply]
That is useful. I'll see if I can add a new setting for that. Well done.

Not sure if this will be today.

Also not sure if this should be an arbitrary value 0-100 like a temp setting or buttons for 20/40/60/80/100 like a fan control.

**Reply by MUN0X** [reply]
No pressure to get this done today :)
Personally I'm already quite happy with the knowledge that I can issue this command to my units via faikin and will probably use home assistant to do that anyway.
The possibility to issue it via a dedicated setting/command that accepts an integer between 40 and 100 would be greatly appreciated though. (The original daikin modules accept 40 as a minumum, not sure if we can go lower with a direct S21 command)

There is an **important disclaimer** with the demand control parameter from what I saw with some testing.
If you have a multi-split, the outdoor unit appears to respect the lowest `max_pow` value set by any of the internal units.
So if you set one of the units to 40%, changing the max_pow of another unit will not impact power draw of the outdoor unit until you raise the 40% limit set by the first unit.
The original daikin-app deals with this by setting the `max_pow` simultaneously on all units that are registered in the app. Effectively synchronizing the setting across all internal units and that's what I've been doing with Home assistant as well.
However I could only set this on 3 of my indoor units and the 4th unit, which was cloud based, just respected the set limit. So the synchronizing is not mandatory, just good practice.

**Reply by revk** [reply]
Today has been mental, so will look tomorrow.

**Reply by Raspdomo** [reply]
@MUN0X : "_If you have a multi-split, the outdoor unit appears to respect the LOWEST max_pow value set by any of the internal units._"
That is exactly the opposite of what Daikin documents: "_if one indoor unit is set on 80% and another is set on 50%, then the outdoor unit will work on 80% capacity_" (free translation from Dutch :-).
Also, in my experience the Daikin Onecta app does not set the max_pow on all units that are registered in the app. I have to set both en_demand and max_pow separately on each unit.

I tend to believe your real life observations more than the often incomplete and unclear Daikin doc, but still: are you sure ?

Me too I have one cloud-only indoor unit on my multisplit, and it would be so nice if it would just "follow" the units that I can control through LAN (kind of slave mode) - or at least not impose its demand control settings over the other units. Otherwise there's no use of implementing this in my home automation system (Domoticz on Raspberry) - though I'd very very much like to !

**Reply by MUN0X** [reply]
I'm pretty sure that how it used to work. However, your "vrije vertaling" might be correct in the meantime.
You see back when i set up my system the Onecta app was brand new. Back then the onecta app was used to setup the new cloud-based units while the older local controllable could be set up via the now obsolete "Daikin online controller" app.  

The old app had demand control as a global setting in the app. Upon changing it the app would send a command to every unit that was registered in the app. Of this I'm 100% sure, I determined this by snooping the wifi traffic when toggling the setting. That's how i got the urls/commands that I gave in one of my initial posts.
The onecta app didn't even support the demand control functionality at that time. 
So your last question was also my concern at the time. But yes, when you turn off "demand control" for the cloud unit it will respect the limit that is set by the LAN controllable units. In fact that's how I've been running the system for years now. (from my experience one of the LAN units must be briefly switched on for a change to the demand control, sent to the LAN controllers, to be applied when only the cloud-based unit is running)  

This lead me to believe that the units would respect the lowest setting. (I remembered i tested this back then and that theory held up as far as I remember). In the end I kept using the logic of the old app via home assistant and just applied the demand control on all LAN controllable units simultaneously. 

In the meantime a lot has changed. The Onecta app now contains the option to change the demand control. And the older LAN controllers have received multiple firmware updates making them compatible with the Onecta app. In the onecta app you can now change the demand control of the old units individually. Perhaps these firmware updates changed the behavior of the LAN controllers when setting demand control?

**Reply by Raspdomo** [reply]
Thanks for your elaborate answer ! I'll test it for myself and report back.

In my home automation system it's the only missing functionality - after that I can get rid of Onecta. I don't want to lose the local LAN API so I don't want any further firmware updates (these come via Onecta).

Anyway, Faikin seems to be a great alternative whenever needed. @revk: many many thanks to share Faikin with the rest of the world !!!

**Reply by Raspdomo** [reply]
I can confirm that my cloud-only unit with demand control set to off does not impose its setting on the whole system. I have to add that my cloud-only unit is switched off (don't know what would happen when it is switched on) - but that doesn't bother me as I rarely use it. The demand control setting on the non-cloud units are taken into account by the outdoor unit ! For me that unblocks the last step of my automation - thanks to @MUN0X !

_Just for information_
Whilst testing the demand control automation I bumped into the following. In Onecta the demand control can be set manually in steps of 5%. 
The set_demand_control max_pow parameter allows any value (within range), the indoor unit accepts it as can be demonstrated with get_demand_control. However if the value is not a multiple of 5 (e.g. 77) the demand control setting of the indoor unit simply disappears completely in Onecta (I admit I that this got me got a little worried :-) ! It reappears when a multiple of 5 is set (e.g. 75) with a set_demand_control.
To me this seems a quirk in the Onecta user interface.

**Comment by turbolooser** [comment]
That sounds awesome. Great work

**Reply by revk** [reply]
Change your `otahost`, e.g. `setting/Faikin/otahost excalibur.belmont.cymru` and upgrade, to get the test version. I should have one to try in a few minutes.

**Comment by revk** [comment]
OK, test release ready for people to try.

**Reply by turbolooser** [reply]
awesome thank you
![image](https://github.com/revk/ESP32-Faikin/assets/115738363/137414ed-22aa-4bac-90a4-26968cbaf3a5)

**Reply by revk** [reply]
The legacy URL should also work.

**Reply by MUN0X** [reply]
Great! I’ll try it when i get home from work.

**Reply by turbolooser** [reply]
I have just switched back to the otahost: ota.revk.uk and do not have the demand feature anymore

**Reply by revk** [reply]
Makes sense, as it seems OK, I'll issue code on `ota.revk.uk` now.

**Reply by revk** [reply]
Done, try now.

**Reply by MUN0X** [reply]
Just tested it and it appears to be working flawless. Thank you so much for implementing this!

Btw, is it possible to set this via mqtt?

**Reply by revk** [reply]
I'll take a look, tomorrow. But it may already be possible as a setting called "demand".

**Reply by turbolooser** [reply]
Would be great to also have this as sensor in HA. Not sure what is required for that...right now I cannot see any entity like demand. I do see it in mqtt but not sure how to get this as sensor in HA. Might require a manual template sensor?

update: I am not able to set "demand":40 using mqtt. @MUN0X have you been succesful?

**Reply by JeanOlivier** [reply]
I just updated using the normal url and I don't see the demand option. It that normal?

Would it be possible to figure out "Powerful" mode with he same approach?

Thanks all!

**Reply by revk** [reply]
We do "powerful".

You may need to set debug or dump mode to see what messages you have. You only get the option if your unit responds to that message.

**Reply by JeanOlivier** [reply]
If you recall I'm the one that can "read" powerful, but apparently can't send the command.

**Reply by MUN0X** [reply]
No, the unit doesn't respond to “demand” via mqtt

**Reply by revk** [reply]
I'll look in the morning...

**Reply by revk** [reply]
OK I have added `demand` with argument 0-100 - new code in a few minutes.

**Comment by apos-t** [comment]
Hi,

thank you for adding this feature. Now I can disable the Daikin Wifi Modules in the future. 
It would be great, if you I can publish the demand control in home assistant as a SelectEntity

There is also an off and auto mode in the app for demand control and also in the cloud integration. A screenshot attached, how it is looking right now in home assistant - with this integration: https://github.com/jwillemsen/daikin_residential_altherma
<img width="310" alt="demand control" src="https://github.com/revk/ESP32-Faikin/assets/109474219/0e75ed5c-8d93-4d53-8cf2-fcb8299b193b">


A second point is, in my opinion the values for demand control could be different for different outdoor units. Is it possible to debug which values are accepted by the outdoor unit? I could set my 4MXM80 to 30% via the web gui and the power consumption is much lower as typically at 40%, which was the minimum at the cloud implementation.

**Reply by revk** [reply]
Can you confirm the exact config JSON for that, and I'll add.

**Reply by apos-t** [reply]
Its the select.mqtt function

# Example configuration.yaml entry
mqtt:
  - select:
      command_topic: topic
      name: "Test Select"
      options:
        - "Option 1"
        - "Option 2"

https://www.home-assistant.io/integrations/select.mqtt/

The create should work with "homeassistant/select/%s/config"

I hope this helps

**Reply by revk** [reply]
I need to know the JSON config I have to send on MQTT.

**Reply by apos-t** [reply]
Example would look like this

Publishing to:

homeassistant/select/demand_office/config

{
    "name": "Demand Control Office",
    "unique_id": "demand_office",
    "command_topic": "homeassistant/select/demand_office/select",
    "options": ["30", "35", "40", "45"] -> and so on
}

**Reply by revk** [reply]
Will look in the morning.

**Reply by revk** [reply]
Might be tomorrow.

**Reply by apos-t** [reply]
no problem, you're doing a great job with this project. I bought 7 Modules and can now switch off the Daikin wifi modules and ban the cloud :-).

**Reply by revk** [reply]
OK not sure if I have this, but set `otahost` to `test.revk.uk` and upgrade - see if that looks sensible.

**Reply by turbolooser** [reply]
Does "ota.revk.uk" (beta) offer the same version or do I need to switch to "test.revk.uk" ?
Maybe a field on the webgui to enter the otahost would be helpful also :-)

**Reply by revk** [reply]
The beta should be pretty up to date.

**Reply by turbolooser** [reply]
So the beta has the ability to set demand control via mqtt also?
Any example on the payload please?

**Reply by revk** [reply]
Err, a number between `0` and `100`

**Reply by turbolooser** [reply]
I guess I need to wait as I cannot get this to work. It might be me as I am not sure which topic and payload to use. Would be much easier to have a "demand" sensor which will be published to HA. Not sure how complex this is.
Thanks for your great work @revk 

@apos-t Maybe you can test this?

**Reply by revk** [reply]
OK I'll do a new beta later today. For HA I need to know the proper JSON config.

**Reply by turbolooser** [reply]
like this?
https://github.com/revk/ESP32-Faikin/discussions/179#discussioncomment-8260515

maybe @apos-t could be more help to this than I am

**Reply by revk** [reply]
Yes, and no. 

`homeassistant/select/demand_office/select` is obviously not the command.

How do I tell it how to send the option?

How do I tell it the status to watch for the status, or is this "command" only?

Ideally I'l love to have an exact JSON (in which I can put the right MAC)... I don't have HA set up, so it is guess work on my part.

**Reply by turbolooser** [reply]
I would love to help on this but would need detailed guidance to do so. 

Which topic needs to be specified to set the "demand" using mqtt?

This is the MAC topic for my faikin-wohnzimmer unit:
{"online":true,"target":19.00,"temp":17.57,"outside":6.50,"liquid":27.00,"comp":24,"demand":50,"fanfreq":11.5,"bletemp":17.57,"blehum":54.46,"blebat":100,"mode":"heat","fan":"auto","swing":"off","preset":"home"}

**Reply by revk** [reply]
Yeh, that is where I am a little stuck. Sounds like doing it as a "command" is wrong, it needs to be a setting like temperature, but maybe HA does not understand it? I'm away for a few days.

**Reply by turbolooser** [reply]
In the cloud addon for the Daikin AC (https://github.com/revk/ESP32-Faikin/discussions/179#discussioncomment-8255197) its a select value as it can only be selected from "40,45,50,55,60,65,70,75,80,85,90,95,100,auto,off". This can be selected only and not entered as Input value.
I assume thats why @apos-t mentioned this select/... earlier.

So hopefully he could help us a little more with this. Thats the only missing part for me to shutoff the cloud integration 👍

**Reply by apos-t** [reply]
sorry, I was at a business trip...

{
"name": "Demand Control Office",
"unique_id": "demand_office",
"command_topic": "homeassistant/select/demand_office/select",
"options": ["30", "35", "40", "45"] -> and so on
}

with the options you publish the values for the select field which you can update with the command_topic from faikin and home assistant. In the app you can select only 5% steps, I think this should be enough. Maybe a couple of the units can go lower than 40%. Therefore I would suggest to start with 30% and go up to 100% in 5% steps like in the screenshot above.

Here is another Json sample from the HA Forum:

https://community.home-assistant.io/t/solved-how-to-get-mqtt-select-in-discovery/349328/2

**Reply by revk** [reply]
So does it just send the topic with the option as the payload?

And no way to report current value back to HA

**Reply by apos-t** [reply]
With this json you create a select field with the options

<img width="514" alt="Bildschirmfoto 2024-02-02 um 18 35 43" src="https://github.com/revk/ESP32-Faikin/assets/109474219/7c102db6-a23f-4b8d-aff7-055fb83aa554">

homeassistant/select/demand_office/config

{
"name": "Demand Control Office",
"unique_id": "demand_office",
"command_topic": "homeassistant/select/demand_office/select",
"options": ["30", "35", "40", "45"] 
}

When the config is done, you can use the topic "homeassistant/select/demand_office/select" to sync the state. When the Value in HA changes you will receive an update at this topic

**Reply by revk** [reply]
I am confused, surely I should be telling it the topic to set it in the Faikin, not homeassistant/select/demand_office/select.

Sorry if I am being thick here.

I'll issue a beta in a moment that has something, see if it is any help (github being laggy so will be a few minutes).

**Reply by turbolooser** [reply]
Could you share what the mqtt topic/payload should be to set demand control on the faikin?
I am still wondering which mqtt command should set the demand control on the faikin..or is it a setting? like the advanced settings?

**Comment by turbolooser** [comment]
From my site I would love to see demand control as a sensor here.
<img width="254" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/a385ea89-563a-4757-8d2e-de567f0ea92d">

**Reply by revk** [reply]
Well I can make as a sensor and a command I guess. Any clues on icon, and unit, and so on? Ideally the exact JSON HA config I need!

**Reply by turbolooser** [reply]
Unit should be %
range will be 30% up to 100% (default in daikin app is 40% up to 100%)
Symbol could be the same than the comp symbol or anything energy related
And a command would be awesome :-)

setting and tracking this "demand" (as sensor) would be really helpful as it seems to be key on the power consumption

**Reply by revk** [reply]
Latest beta might have working command side, can you try?

**Reply by MUN0X** [reply]
I get the following error when i try to set demand via the selection box
![image](https://github.com/revk/ESP32-Faikin/assets/78272516/de2802e5-afab-4df5-84d6-e20b8531d831)

**Reply by revk** [reply]
I don't know enough about HA. So suggestions welcome...

**Reply by JeanOlivier** [reply]
I'm no expert either but from that message I'd try to replace 0 by "0".

**Reply by revk** [reply]
I'll give that a go in the morning.

**Reply by revk** [reply]
OK, issuing a beta now with that change, do try.

**Reply by turbolooser** [reply]
what should be the command to set the demand control?

@JeanOlivier --> 0 is not a valid option for demand control I guess. Its either auto,off, or 40, 45,50 etc until 100

**Reply by revk** [reply]
`command/Faikin/demand 40`

or mac address `/demand 40`

**Reply by turbolooser** [reply]
My bad... I have found the entity now on HA :-) 

so its here
<img width="264" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/b0b00ad7-791e-40b3-9678-06f87b910283">

But when I select a value from the list it does not change anything on the faikin

**Reply by turbolooser** [reply]
Ok...we are getting there now I hope.

I have been able to set the demand using mqtt now the first time ever 👍 
topic: "command/faikin-schlafzimmer/demand"
payload: "40"

It was accepted by the faikin.
topic: command/308398D3D814/demand
payload: 40
was also accepted

**Reply by turbolooser** [reply]
Now we need to figure out how to send this command to the faikin from HA when I select something in the list of options in HA so that we could select it from HA directly

**Reply by turbolooser** [reply]
To me it looks like that when I select an option in the List in HA there is no command issues to the faikin module. I do not see anything on the mqtt broker debug logs on HA...so I assume that no command is being sent from HA to the Faikin

When I run the mqtt command directly I do see the command on the topic correctly BUT.... it seems that the value gets reverted to the older value after a couple of seconds. So I set it to 30... and then it reverts back to the previous value of 65 after 8-10 seconds

<img width="183" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/9faedeb4-28d0-4841-8265-1e13298fe07a">

**Reply by turbolooser** [reply]
Any idea on why there is no command send to the Faikin?

**Reply by revk** [reply]
I don't know enough about HA. Others have reported this working I believe.

**Reply by turbolooser** [reply]
I have not seen any confirmation from HA. Pure mqtt works but value gets reverted (maybe by cloud integration). 
Anyone has this working from HA select entity?

@MUN0X ?

**Reply by MUN0X** [reply]
Setting the demand via mqtt works for me. (At least it did a few fw versions ago, havent checked recently.) Does setting it via the web interface work for you?

**Comment by revk** [comment]
Values reverted means the S21 is not confirming back the setting. It may be that it does not allow some values, or even allow the setting.

**Comment by apos-t** [comment]
Hi, 
I was away for a couple of days and could now do some testing, it's not working at the moment.

 I forgot one point in the json, the state_topic... to receive updates from faking in HA

{
"name": "Demand Control Office",
"unique_id": "demand_office",
"command_topic": "homeassistant/select/demand_office/select", **-> this is the topic faikin needs to subscribe for updates from HA, when the value changes.**
"state_topic": "homeassistant/select/demand_office", -> ** this topic is for faikin to send / set the Value in HA**
"options": ["30", "35", "40", "45"] 
}

**Reply by revk** [reply]
Ok leave with me.

**Reply by turbolooser** [reply]
Please add the options "off" and "auto" also as they do exist in the Daoikin cloud options
Thank you

**Reply by revk** [reply]
The various messages I have seen do not suggest these are separate messages.

My understanding is auto is 100%. And off is 100% but when on it remembers what you set. Though auto may be a thing they do for the cloud or something, so would need a spec of that.

Feel free to provide S21 messages otherwise.

**Reply by turbolooser** [reply]
There is a lot of discussion on what "auto" means and does. :-D Seems that in the cloud daikin is trying to use the values from last day if set to "auto" but who knows

And if "off" means 100% is also not clear. I can see that If I set to 100% its not the same power consumption than "off"
But not sure if that is a general pattern or its just my install

**Reply by MUN0X** [reply]
From what I’ve seen:
off = 100%
Auto = either 100% or 85%. I suspect the original daikin module chooses one based on another parameter. Be it deviation from setpoint, time of day, outside temperature,… no clue.

**Reply by MUN0X** [reply]
Set it to 100% and then turn the unit off and back on. You should get full power again

**Reply by apos-t** [reply]
@revk Do you need any other Input for the implementation?

**Reply by revk** [reply]
What are we waiting for at the moment, I thought we had it working.

**Reply by thomaslunze** [reply]
I can only set the deman control using mqtt messages like:

"command/faikin-wohnzimmer/demand" = 50 (payload)

but I cannot get it working with any select in HA using the sensor which is 
![image](https://github.com/revk/ESP32-Faikin/assets/69297495/ac54638f-1795-49d5-a5ef-876dcd4a87c1)
 It just does not do anything for me in HA

**Reply by revk** [reply]
Are you on latest code?

Can you log MQTT messages that HA sends?

Can you see anything wrong with the HA config we are registering?

**Reply by thomaslunze** [reply]
When I select "85" it goes like this
![image](https://github.com/revk/ESP32-Faikin/assets/69297495/45115a2a-202d-4599-8de0-cea6014742b4)
but it does not do anything on the webgui of the faikin

when i run the mqtt command which sets the demand correctly its like this
![image](https://github.com/revk/ESP32-Faikin/assets/69297495/63fb4fc4-6148-4a9f-bb6c-33b179f4e47c)

**Reply by revk** [reply]
That first one should work. Let me double check.

Ah, I may see it. One sec. I'll do another beta.

OK, try latest beta.

**Comment by apos-t** [comment]
you set the command topic to:
 jo_stringf (j, "cmd_t", "%s/demand", revk_id);

that's the topic HA is sending the updates

In my case  the message from HA looks like this:

Topic: 0C8B9596ABC0/demand
Value: 50

**Reply by revk** [reply]
Yes, I have changed it now in the latest beta, hence suggesting you try it, 16 minutes ago.

**Reply by apos-t** [reply]
Sorry, I was too slow. On the latest beta it's working. Thank you!

**Reply by revk** [reply]
There are some people that file bug reports just after I have issued a fix... So well done getting it in the right order :-)

**Reply by thomaslunze** [reply]
I can confirm its working now. awesome

**Comment by eupesco** [comment]
Hi, 
I just came across a strange thing: when I change the demand control in Faikin, it obviously works (because the power of the HVAC is increasing/decreasing). But when I look in the Onecta app, the demand control value does not change. Has anyone seen this behavior? 
I remember that the HA Daikin integration (in HACS, not the official one) was able to change the demand control (before it was changed because of the cloud change at Daikin) and the value was synchronized with the Onecta app.

**Reply by wurmtal868** [reply]
The Onecta seems to push the demand value but is not able to read it back. That's my experience with my two multi split systems.
You can always see the demand value on the Faikin GUI
